pkg/proxy/iptables/proxier_test.go

```diff
@@ -4391,6 +4348,135 @@ COMMIT
 -A KUBE-XLB-AQI2S6QIMU7PVVRP -m comment --comment "ns1/svc1 has no local endpoints" -j KUBE-MARK-DROP
 -A KUBE-SERVICES -m comment --comment "kubernetes service nodeports; NOTE: this must be the last rule in this chain" -m addrtype --dst-type LOCAL -j KUBE-NODEPORTS
 COMMIT
+`,
+		},
+		{
+			name:                   "ProxyTerminatingEndpoints enabled, terminating endpoints on remote node",
+			terminatingFeatureGate: true,
+			endpointslice: &discovery.EndpointSlice{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      fmt.Sprintf("%s-1", "svc1"),
+					Namespace: "ns1",
+					Labels:    map[string]string{discovery.LabelServiceName: "svc1"},
+				},
+				Ports: []discovery.EndpointPort{{
+					Name:     utilpointer.StringPtr(""),
+					Port:     utilpointer.Int32Ptr(80),
+					Protocol: &tcpProtocol,
+				}},
+				AddressType: discovery.AddressTypeIPv4,
+				Endpoints: []discovery.Endpoint{
+					{
+						// this endpoint won't be used because it's not local,
```

Comment:
no, we're intentionally including an endpoint that we want it to ignore, to make sure that it ignores it

Reply:
Shouldn't this get a REJECT?

Reply:
No, we only generate a REJECT rule if there are no endpoints at all. For a Local traffic policy service with remote endpoints but no local endpoints, we just DROP traffic, because if the LB is lagging slightly behind kube-proxy and forwards a connection to this node after we stop listening, we don't want to cause a "connection refused" to get sent back to the client. So we just drop the packet, and the client will eventually retry its connection to the load balancer, and the load balancer will (hopefully) send to another node.

I can add more comments to the proxier code explaining this if you want?

Reply:
(That was always the expected behavior, and is how the code worked pre-terminating-endpoints, but we didn't have test cases for all parts of it.)

Reply:
If we don't REJECT it, the packet falls off the bottom and can actually hit the network, right?  ISTR this was a problem in the past but I don't recall.  Either way, having a timeout seems bad?

Reply:
is not dropped by?
> -A KUBE-FW-AQI2S6QIMU7PVVRP -m comment --comment "ns1/svc1 loadbalancer IP" -j KUBE-MARK-DROP

Reply:
I missed that.

Reply:
It still feels weird to me that we have inconsistent behavior in these cases.  DROP feels rude.  Is there a reason not to reject?